src/api/async_element.rs

```diff
@@ -352,6 +352,44 @@ mod tests {
         assert_eq!(router_output.len(), 2001);
     }
 
+    #[test]
+    fn one_async_element_with_empty_stream() {
```

Comment:
This is the right idea. We have an issue open to abstract away some of the tokio running and assertion code away, but you'll necessarily have to define the input stream for your router, configure it via Element links, and assert that some stuff happened to the output stream.

Reply:
This looks good to me, does what we expect.